### PR TITLE
Update Tinycrypt for mbedTLS 3.x support

### PR DESCRIPTION
**Pull request by rosahay-silabs**
#### Problem / Feature
What is being fixed? What is the feature being added? Examples:
* tinycrypt was integrated with mbedTLS 2.x
* due to recent PSRIT, mbedTLS 3.x was recommended to be integrated.

#### Change overview
* Remove unused source from tinycrypt.
* Adding additional changes to existing tinycrypt source to fix compilation issues.
* Restyling the source files as maintained in armbed barebone source.

#### Testing
* Manually tested with SiWx917

### CODE REVIEW

File: matter/mbedtls/tinycrypt/src/ecc.c
Comment (mkardous-silabs) [inline matter/mbedtls/tinycrypt/src/ecc.c:1]: what was changed in this file? Hard to tell from the diff

File: matter/mbedtls/tinycrypt/src/ecc_dsa.c
Comment (mkardous-silabs) [inline matter/mbedtls/tinycrypt/src/ecc_dsa.c:1]: same question

File: matter/mbedtls/tinycrypt/src/tinycrypt_util.c
Comment (mkardous-silabs) [inline matter/mbedtls/tinycrypt/src/tinycrypt_util.c:1]: where were these function taken from?

File: matter/si91x/siwx917/BRD4342A/autogen/sli_mbedtls_config_autogen.h [hunk 1]
```diff
@@ -12,6 +12,7 @@
 #define MBEDTLS_ASN1_WRITE_C
 #define MBEDTLS_BASE64_C
 #define MBEDTLS_CIPHER_C
+#define MBEDTLS_CIPHER_PADDING_PKCS7
```
Comment (mkardous-silabs) [inline matter/si91x/siwx917/BRD4342A/autogen/sli_mbedtls_config_autogen.h:15]: can we revert this change since the define was already present?

File: matter/si91x/siwx917/BRD4343A/autogen/sli_mbedtls_config_autogen.h [hunk 2]
```diff
@@ -12,6 +12,7 @@
 #define MBEDTLS_ASN1_WRITE_C
 #define MBEDTLS_BASE64_C
 #define MBEDTLS_CIPHER_C
+#define MBEDTLS_CIPHER_PADDING_PKCS7
```
Comment (mkardous-silabs) [inline matter/si91x/siwx917/BRD4343A/autogen/sli_mbedtls_config_autogen.h:15]: can we revert this change since the define was already present?

File: matter/mbedtls/tinycrypt/inc/tinycrypt/tinycrypt_util.h [hunk 3]
```diff
@@ -0,0 +1,17 @@
+#ifndef TINYCRYPT_PLATFORM_UTIL_H
+#define TINYCRYPT_PLATFORM_UTIL_H
+
+#include <stddef.h>
+#include <stdint.h>
+
+#if defined(SL_MBEDTLS_USE_TINYCRYPT)
```
[the remark this answers is not in the document]
Comment (rosahay-silabs) [reply matter/mbedtls/tinycrypt/inc/tinycrypt/tinycrypt_util.h:7]: Here I guess we can remove, but keeping it in other places to align with the original source found https://fuchsia.googlesource.com/third_party/github.com/ARMmbed/mbedtls/+/refs/heads/upstream/baremetal/tinycrypt/

File: matter/mbedtls/tinycrypt/inc/tinycrypt/tinycrypt_util.h [hunk 4]
```diff
@@ -0,0 +1,17 @@
...
+void mbedtls_platform_random_buf(uint8_t *buf, size_t len);
+void *mbedtls_platform_memset(void *, int, size_t);
+void *mbedtls_platform_memcpy(void *dst, const void *src, size_t num);
+void mbedtls_platform_random_delay(void);
+uint32_t mbedtls_platform_random_in_range(uint32_t num);
+// int mbedtls_platform_memmove(void *dst, const void *src, size_t num);
+// int mbedtls_platform_memequal(const void *buf1, const void *buf2, size_t num);
```
[the remark this answers is not in the document]
Comment (rosahay-silabs) [reply matter/mbedtls/tinycrypt/inc/tinycrypt/tinycrypt_util.h:14]: Yes, it should be deleted.

File: matter/mbedtls/tinycrypt/src/ecc.c [hunk 5]
```diff
@@ -1,1738 +1,1869 @@
...
+ *  POSSIBILITY OF SUCH DAMAGE.
+ */
+
+#if !defined(MBEDTLS_CONFIG_FILE)
+#include "mbedtls/config.h"
+#else
+#include MBEDTLS_CONFIG_FILE
+#endif
+
+#if defined(SL_MBEDTLS_USE_TINYCRYPT)
```
[the remark this answers is not in the document]
Comment (rosahay-silabs) [reply matter/mbedtls/tinycrypt/src/ecc.c:66]: https://fuchsia.googlesource.com/third_party/github.com/ARMmbed/mbedtls/+/refs/heads/upstream/baremetal/tinycrypt/

aligning with the source code, I am sure the config will play a part in case the sources get included without the MACRO.

File: matter/si91x/siwx917/BRD4342A/autogen/sli_mbedtls_config_autogen.h (hunk 1 above)
Comment (rosahay-silabs) [reply matter/si91x/siwx917/BRD4342A/autogen/sli_mbedtls_config_autogen.h:15]: this was present for only PSA and I moved into the common configuration for both PSA and tinycrypt

File: matter/si91x/siwx917/BRD4343A/autogen/sli_mbedtls_config_autogen.h (hunk 2 above)
Comment (rosahay-silabs) [reply matter/si91x/siwx917/BRD4343A/autogen/sli_mbedtls_config_autogen.h:15]: this was present for only PSA and I moved into the common configuration for both PSA and tinycrypt

File: matter/mbedtls/tinycrypt/src/tinycrypt_util.c
Comment (rosahay-silabs) [reply matter/mbedtls/tinycrypt/src/tinycrypt_util.c:1]: This file is unchanged from your commits. I just added some fixes for compilation by including headers.

File: matter/mbedtls/tinycrypt/src/ecc.c [hunk 6]
```diff
@@ -1,1738 +1,1869 @@
...
+#include "mbedtls/sha256.h"
+
+#if defined MBEDTLS_OPTIMIZE_TINYCRYPT_ASM
+#ifndef asm
+#define asm __asm
+#endif
+#endif /* MBEDTLS_OPTIMIZE_TINYCRYPT_ASM */
+
+#if (MBEDTLS_VERSION_NUMBER >= 0x03000000)
+#include "mbedtls/compat-2.x.h"
```
Comment (rosahay-silabs) [inline matter/mbedtls/tinycrypt/src/ecc.c:82]: change

**Review by rcasallas-silabs — APPROVED** [review]

File: matter/mbedtls/tinycrypt/src/ecc.c (hunk 6 above)
Comment (mkardous-silabs) [reply matter/mbedtls/tinycrypt/src/ecc.c:82]: why did we need to add this since we would use mbedtls 3.0?

**Review by mkardous-silabs — APPROVED** [review]

File: matter/mbedtls/tinycrypt/src/ecc.c (hunk 6 above)
Comment (rosahay-silabs) [reply matter/mbedtls/tinycrypt/src/ecc.c:82]: Adding this to maintain backward compatibility since this can be used for mbedTLS 2.x as well then
